modules/home-environment.nix

```
@@ -592,10 +592,21 @@ in
         ''
       else
         ''
+          function hm-internal-replace-profile {
+
+            nix profile list \
+              | { grep 'home-manager-path$' || test $? = 1; } \
+              | awk -F ' ' '{ print $4 }' \
+              | cut -d ' ' -f 4 \
+              | xargs -t $DRY_RUN_CMD nix profile remove $VERBOSE_ARG
+
+            $DRY_RUN_CMD nix profile install $1
+          }
+
           if [[ -e "$nixProfilePath"/manifest.json ]] ; then
-            INSTALL_CMD="nix profile install"
+            INSTALL_CMD="hm-internal-replace-profile"
```

This will cause the error message in the `if` statement below to be incorrect for the user.

---

To clarify, I mean the message that shows up in the line starting with `_iError` below.

I think to fix this one would need something like

```suggestion
            INSTALL_CMD="hm-internal-replace-profile"
            INSTALL_CMD_ACTUAL=nixReplaceProfile
```

and use `$INSTALL_CMD_ACTUAL` instead of `$INSTALL_CMD` in

``` sh
if ! $DRY_RUN_CMD $INSTALL_CMD ${cfg.path} ; then
```

If you solve it like this, then don't forget to also add

``` sh
INSTALL_CMD_ACTUAL="nix-env -i"
```

for the non-experimental code path.